query/src/interpreters/interpreter_table_create_v2.rs

```diff
@@ -0,0 +1,187 @@
+            .iter()
+            .all(|table| table.name() != self.plan.table.as_str());
+
+        let engine_desc: Option<StorageDescription> = catalog
+            .get_table_engines()
+            .iter()
+            .find(|desc| {
+                desc.engine_name.to_string().to_lowercase() == engine.to_string().to_lowercase()
+            })
+            .cloned();
+
+        match engine_desc {
+            Some(engine) => {
+                if !self.plan.cluster_keys.is_empty() && !engine.support_cluster_key {
+                    return Err(ErrorCode::UnsupportedEngineParams(format!(
+                        "Unsupported cluster key for engine: {}",
+                        engine.engine_name
+                    )));
+                }
+            }
+            None => {
+                if name_not_duplicate {
+                    return Err(ErrorCode::UnknownTableEngine(format!(
+                        "Unknown table engine {}",
+                        engine
+                    )));
+                }
+            }
+        }
+
+        match &self.plan.as_select {
+            Some(select_plan_node) => {
+                self.create_table_as_select(input_stream, select_plan_node.clone())
+                    .await
+            }
+            None => self.create_table().await,
+        }
+    }
+}
+
+impl CreateTableInterpreterV2 {
+    async fn create_table_as_select(
+        &self,
+        input_stream: Option<SendableDataBlockStream>,
+        select_plan: Box<Plan>,
+    ) -> Result<SendableDataBlockStream> {
+        let tenant = self.ctx.get_tenant();
+        let catalog = self.ctx.get_catalog(&self.plan.catalog)?;
+
+        // TODO: maybe the table creation and insertion should be a transaction, but it may require create_table support 2pc.
+        catalog.create_table(self.plan.clone().into()).await?;
+        let table = catalog
+            .get_table(tenant.as_str(), &self.plan.database, &self.plan.table)
+            .await?;
+
+        // If the table creation query contains column definitions, like 'CREATE TABLE t1(a int) AS SELECT * from t2',
+        // we use the definitions to create the table schema. It may happen that the "AS SELECT" query's schema doesn't
+        // match the table's schema. For example,
+        //
+        //   mysql> create table t2(a int, b int);
+        //   mysql> create table t1(x string, y string) as select * from t2;
+        //
+        // For the situation above, we implicitly cast the data type when inserting data.
+        // The casting and schema checking is in interpreter_insert.rs, function check_schema_cast.
+        let table_schema = table.schema();
+        let select_fields: Vec<DataField> = select_plan
+            .schema()
+            .fields()
+            .iter()
+            .filter_map(|f| table_schema.field_with_name(f.name()).ok())
+            .cloned()
+            .collect();
+        let schema = DataSchemaRefExt::create(select_fields);
```

Comment:
Can we enable omitting the schema like:

```
> create table tt as select * from t1;

Elapsed: 11 ms

```

Reply:
> Can we enable omitting the schema like:
> 
> ```
> > create table tt as select * from t1;
> 
> Elapsed: 11 ms
> ```

1.
```sql
create table tt as select * from t1;
```
here, we can use the `table_schema`(line 148) and omit the `schema`(line 156).


2.
```sql
create table tt (i INT) as select * from t1
```
the `table_schema` is different from the `schema` ,  so it can't be omitted.

Reply:
Need a stateless test to cover:

```
create table tt as select * from t1;
```

Reply:
I'll add it later in #6648